src/test/scala/edu/berkeley/cs/rise/opaque/OpaqueOperatorTests.scala

```diff
@@ -582,6 +589,20 @@ trait OpaqueOperatorTests extends FunSuite with BeforeAndAfterAll { self =>
     df.select(dot($"v1", $"v2")).collect
   }
 
+  testAgainstSpark("upper") { securityLevel =>
```

Comment:
I'm curious how this test passes... could you paste the logical and physical plan queries here?

Reply:
Also, the test should test for NULL values as well.

Reply:
scala> dfe.select(upper($"v1")).explain
== Physical Plan ==
EncryptedProject [upper(v1#4) AS upper(v1)#26]
+- EncryptedLocalTableScan [v1#4, v2#5], [[lower,upper]]

Reply:
Still working on adding the null test...

Reply:
Maybe you can create two tests, one for nulls and one for non-nulls

Reply:
Added two tests - one including null and one without